src/librustc_mir/transform/const_prop.rs

```diff
@@ -497,6 +497,67 @@ impl<'a, 'mir, 'tcx> ConstPropagator<'a, 'mir, 'tcx> {
             },
         }
     }
+
+    fn replace_with_const(&self, rval: &mut Box<Rvalue<'tcx>>, value: Const<'tcx>) {
+        let operand = value;
+        if let interpret::Operand::Immediate(im) = *operand {
+            match im {
+                interpret::Immediate::Scalar(ScalarMaybeUndef::Scalar(scalar)) => {
+                    *rval = Box::new(Rvalue::Use(Operand::Constant(Box::new(
```

Comment:
create a helper function for obtaining an `Operand` from a `ty::Const` (or maybe even from a `Scalar` + `Ty`, although in the future we'll probably need the `ty::Const`, too)